src/foam/u2/FragmentedTextField.js

```diff
@@ -95,19 +97,36 @@ foam.CLASS({
       this.numOfParts = this.delegates.length;
 
       return this;
+    },
+    function processData(data, index, arr) {
+      // returns an array of data chunks in the size of the delegates
+      var currentElement = this.childNodes[index];
+      if ( ! data || ! currentElement ) return arr;
+
+      var maxLength = this.delegates[index].maxLength;
+      arr.push(data.substring(0, maxLength));
+      return this.processData(data.substring(maxLength), index + 2, arr);
     }
   ],
 
   listeners: [
     {
       name: 'onDataUpdate',
       code: function() {
-        var currentElement = this.childNodes[this.currentIndex];
-        if ( currentElement.getAttribute('maxlength') <= currentElement.data.length ) {
-          this.currentIndex = this.currentIndex + 2;
-          this.childNodes[this.currentIndex].focus();
+        var arr = this.processData(this.childNodes[this.currentIndex].data, this.currentIndex, []);
+        for ( var i = 0; i < arr.length; i++ ) {
+          this.childNodes[this.currentIndex].data = arr[i];
+          if ( next = this.childNodes[this.currentIndex + 2] ) {
+            if ( this.delegates[this.currentIndex].maxLength > arr[i].length ) break;
+
+            if ( ! ( next && next.data ) ) {
+              this.currentIndex = this.currentIndex + 2;
```

Comment:
I don't remember why this was `+2`. What happens when it's `++`?

Reply:
the textfields in the delegates are separated by stuff like '-', now I'm wondering if those might be better if added as a suffix in the fragment model.. any thoughts?

Reply:
Oh, or maybe a typecheck would be better. IMO it's a later problem since we're always using it for this situation, but there should be a `// TODO` or something at least

Reply:
yeah makes sense, commented/ticketed 
https://github.com/foam-framework/foam2/pull/4310/files#diff-2849a5a16bce0b829d0e6707e081a15d147abc768ac7c78c7b9cf898355b1c9cR123
https://nanopay.atlassian.net/browse/NP-2744